admin/class-perfecty-push-admin.php

```diff
@@ -861,4 +881,77 @@ public function print_settings_opt_in() {
 			esc_html( $value )
 		);
 	}
+
+	/**
+	 * Get first image URL in post content
+	 *
+	 * @since 1.0.8
+	 *
+	 * @return string $thumbnail_url on success, '' on failure
+	 */
+	public function get_first_image_url( $post )
+	{
+		$content = $post->post_content;
+		$regex = '/src="([^"]*)"/';
+		preg_match_all( $regex, $content, $matches );
+		$matches = array_reverse( $matches );
+		// this is the image url of the first img embedded in content.
+		$img_url = $matches[0][0];
+
+		// this is the image post id.
+		$post_img_id = $this->get_attachment_id( $img_url );
+
+		if ($post_img_id !== 0 ) {
+			// this is an array related to the thumbnail of the first image.
+			$img_thumb_url = wp_get_attachment_image_src( $post_img_id, $size = 'thumbnail', $icon = false );
+		} else {
+			$img_thumb_url = '';
+		}
+
+		// we return the URL of the thumbnail.
+		return $img_thumb_url[0];
```

Comment:
From the `else` clause before, we will be doing  `$img_thumb_url[0]` to an empty string `''`

Reply:
I got it. I will fix it